src/bin.rs

```diff
@@ -145,3 +175,255 @@ fn play(cfg: Play) {
         }
     }
 }
+
+fn stats(cfg: Stats) {
+    let file = File::open(cfg.input).expect("failed to compute statistics for recorded file");
+
+    let (tx, rx): (Sender<TelemetryChannelType>, Receiver<TelemetryChannelType>) =
+        std::sync::mpsc::channel();
+    std::thread::spawn(move || {
+        gather_telemetry_from_file(file, tx);
+    });
+
+    let mut telemetry_messages: Vec<TelemetryMessage> = Vec::new();
+
+    let mut nb_boot_messages: u32 = 0;
+    let mut nb_alarm_traps: u32 = 0;
+    let mut nb_data_snapshots: u32 = 0;
+    let mut nb_machine_state_snapshots: u32 = 0;
+    let mut nb_stopped_messages: u32 = 0;
+
+    loop {
+        match rx.try_recv() {
+            Ok(message) => {
+                match message {
+                    Ok(TelemetryMessage::BootMessage(_)) => {
+                        nb_boot_messages += 1;
+                    }
+                    Ok(TelemetryMessage::AlarmTrap(_)) => {
+                        nb_alarm_traps += 1;
+                    }
+                    Ok(TelemetryMessage::DataSnapshot(_)) => {
+                        nb_data_snapshots += 1;
+                    }
+                    Ok(TelemetryMessage::MachineStateSnapshot(_)) => {
+                        nb_machine_state_snapshots += 1;
+                    }
+                    Ok(TelemetryMessage::StoppedMessage(_)) => {
+                        nb_stopped_messages += 1;
+                    }
+                    _ => {}
+                }
+                telemetry_messages.push(message.unwrap());
```

Comment:
I think I would have put this in the previous match block, inside a big `Ok(msg)` case where I would have handled the various message types `match msg { TelementryMessage::BootMessage(_) => ..., ... }`.
I believe this is always better to avoid `unwrap()` when possible/easy.

Reply:
I don't understand your point. Can we have a call today about this ?